datalad_neuroimaging/extractors/bids_dataset.py

```diff
@@ -0,0 +1,259 @@
+    TASK = 'task'
+    ACQUISITION = 'acquisition'
+    CEAGENT = 'ceagent'
+    RECONSTRUCTION = 'reconstruction'
+    DIRECTION = 'direction'
+    RUN = 'run'
+    PROC = 'proc'
+    MODALITY = 'modality'
+    ECHO = 'echo'
+    FLIP = 'flip'
+    INV = 'inv'
+    MT = 'mt'
+    PART = 'part'
+    RECORDING = 'recording'
+    SPACE = 'space'
+    SUFFIX = 'suffix'
+    SCANS = 'scans'
+    FMAP = 'fmap'
+    DATATYPE = 'datatype'
+    EXTENSION = 'extension'
+
+BIDSCONTEXT = {
+    "@id": "https://doi.org/10.5281/zenodo.3686061",
+    'description': 'ad-hoc vocabulary for the Brain Imaging Data Structure (BIDS) standard v1.6.0',
+    'type': vocabulary_id,
+}
+
+DATASET = 'dataset'
+
+BIDS_PROPERTIES_MAPPING= {
+    BIDSProperties.NAME: BIDSProperties.NAME,
+    BIDSProperties.BIDSVERSION: BIDSProperties.BIDSVERSION,
+    BIDSProperties.DATASETTYPE: BIDSProperties.DATASETTYPE,
+    BIDSProperties.LICENSE: BIDSProperties.LICENSE,
+    BIDSProperties.AUTHORS: BIDSProperties.AUTHORS,
+    BIDSProperties.ACKNOWLEDGEMENTS: BIDSProperties.ACKNOWLEDGEMENTS,
+    BIDSProperties.HOWTOACKNOWLEDGE: BIDSProperties.HOWTOACKNOWLEDGE,
+    BIDSProperties.FUNDING: BIDSProperties.FUNDING,
+    BIDSProperties.ETHICSAPPROVALS: BIDSProperties.ETHICSAPPROVALS,
+    BIDSProperties.REFSANDLINKS: BIDSProperties.REFSANDLINKS,
+    BIDSProperties.DATASETDOI: BIDSProperties.DATASETDOI,
+    BIDSProperties.HEDVERSION: BIDSProperties.HEDVERSION,
+}
+
+BIDS_ENTITIES_MAPPING = {
+    BIDSEntities.SUBJECT: BIDSEntities.SUBJECT,
+    BIDSEntities.SESSION: BIDSEntities.SESSION,
+    BIDSEntities.TASK: BIDSEntities.TASK,
+    BIDSEntities.ACQUISITION: BIDSEntities.ACQUISITION,
+    BIDSEntities.CEAGENT: BIDSEntities.CEAGENT,
+    BIDSEntities.RECONSTRUCTION: BIDSEntities.RECONSTRUCTION,
+    BIDSEntities.DIRECTION: BIDSEntities.DIRECTION,
+    BIDSEntities.RUN: BIDSEntities.RUN,
+    BIDSEntities.PROC: BIDSEntities.PROC,
+    BIDSEntities.MODALITY: BIDSEntities.MODALITY,
+    BIDSEntities.ECHO: BIDSEntities.ECHO,
+    BIDSEntities.FLIP: BIDSEntities.FLIP,
+    BIDSEntities.INV: BIDSEntities.INV,
+    BIDSEntities.MT: BIDSEntities.MT,
+    BIDSEntities.PART: BIDSEntities.PART,
+    BIDSEntities.RECORDING: BIDSEntities.RECORDING,
+    BIDSEntities.SPACE: BIDSEntities.SPACE,
+    BIDSEntities.SUFFIX: BIDSEntities.SUFFIX,
+    BIDSEntities.SCANS: BIDSEntities.SCANS,
+    BIDSEntities.FMAP: BIDSEntities.FMAP,
+    BIDSEntities.DATATYPE: BIDSEntities.DATATYPE,
+    BIDSEntities.EXTENSION: BIDSEntities.EXTENSION,
+}
+
+# Entities for which variable collections can be extracted
+BIDS_COLLECTION_ENTITIES = [BIDSEntities.RUN, BIDSEntities.SESSION, BIDSEntities.SUBJECT, DATASET]
+
+class BIDSDatasetExtractor(DatasetMetadataExtractor):
+    """
+    Top level BIDS extractor class interfacing with metalad
+    Inherits from metalad's DatasetMetadataExtractor class
+    """
+
+    def get_id(self) -> UUID:
+        # Note sure what the process should be for generating these
```

Comment:
It should be a UUID that is not used (most likely) for another metalad-extractor or -filter. For example, you could use a version 4 UUID generated here: `https://www.uuidgenerator.net/`